spec/features/user_can_renew_book_spec.rb

```diff
@@ -0,0 +1,94 @@
+require 'rails_helper'
+require 'date'
+
+feature 'User can renew a checked out book' do
+  scenario 'sucessfully' do
+    patron = create(:patron)
+    book = create(:book)
+    checkout = create(:checkout, user: patron, book: book)
+
+    visit profile_path(as: patron)
+
+    click_on 'Renew'
+
+    checkout.reload
+
+    original_checkout_date = book.checkouts.first.created_at
+    expected_due_date_after_renewal = format_date(original_checkout_date + (2 * Checkout::CHECKOUT_PERIOD_IN_DAYS.days))
+    due_date_after_renewal = format_date(book.checkouts.last.due_on)
+    expect(page).to have_due_date(due_date_after_renewal)
+    expect(due_date_after_renewal).to eq(expected_due_date_after_renewal)
+    expect(page).to have_content(t('checkouts.update.successful_message'))
+  end
+
+  scenario 'prior fines carry over when an overdue book is renewed' do
```

Comment:
I think if you say "carry over"  that implies that fines are moving. Instead I would consider saying "prior fines are included..."